src/common/m_merger.cpp

```diff
@@ -199,3 +199,95 @@ bool Merger::m_receive_select(PCBuffer *pcbuffer) {
 
 	return true;
 }
+
+bool Merger::m_receive_select1(PCBuffer *pcbuffer) {
```

Comment:
this function is to avoid blocking in get_sorted() which is a globally sort function.